corehq/motech/repeaters/expression/forms.py

```diff
@@ -54,47 +68,31 @@ def get_ordered_crispy_form_fields(self):
         ]
 
     def clean_configured_expression(self):
-        try:
-            ExpressionFactory.from_spec(
-                self.cleaned_data['configured_expression'], FactoryContext.empty(domain=self.domain)
-            )
-        except BadSpecError as e:
-            raise ValidationError(e)
-
-        return self.cleaned_data['configured_expression']
+        return self._clean_expression('configured_expression')
 
     def clean_configured_filter(self):
-        try:
-            FilterFactory.from_spec(self.cleaned_data['configured_filter'])
-        except BadSpecError as e:
-            raise ValidationError(e)
-
-        return self.cleaned_data['configured_filter']
+        return self._clean_filter(self.cleaned_data['configured_filter'])
 
     def clean_case_action_expression(self):
-        raw = self.cleaned_data.get('case_action_expression')
-        if raw:
-            try:
-                ExpressionFactory.from_spec(
-                    raw, FactoryContext.empty(domain=self.domain)
-                )
-            except BadSpecError as e:
-                raise ValidationError(e)
-
-        return raw
+        return self._clean_expression('case_action_expression')
 
     def clean_case_action_filter_expression(self):
         raw = self.cleaned_data.get('case_action_filter_expression')
         if raw:
-            try:
-                FilterFactory.from_spec(raw)
-            except BadSpecError as e:
-                raise ValidationError(e)
-
+            return self._clean_filter(raw)
         return raw
 
     def clean(self):
         cleaned_data = super().clean()
+        request_method = self.cleaned_data['request_method']
+        body_required = request_method in (REQUEST_POST, REQUEST_PUT)
```

Comment:
POST and PUT requests [don't need a body](https://stackoverflow.com/questions/4191593/is-it-considered-bad-practice-to-perform-http-post-without-entity-body), but I appreciate that people _expect_ them to have a body, and if the body is missing, the user is _probably_ making a mistake.

Perhaps out of scope for this change, but how do you feel about warning the user instead of raising a ValidationError?

Reply:
I'd prefer not to change that in this PR. I think it's some as is since empty payloads can be accomplished using a 'constant' expression if really necessary.